services/documentation_service.py

```diff
@@ -98,6 +98,44 @@ def create_document_generation_job(
 
         return doc_id
 
+    def update_document_generation_job(
+            self,
+            background_tasks: BackgroundTasks,
+            doc_id: str,
+            model: LlmModelEnum = None
+    ) -> str:
+        doc = self.firebase_client.get_documentation(doc_id)
+        github_url = doc.get("github_url")
+        blob_url = doc.get("bucket_url")
+
+        doc_status = DocsStatusEnum(doc.get("status"))
+
+        if (doc_status == DocsStatusEnum.STARTED):
+            raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail=f"Data is still being generated for this id, so it cannot be regenerated yet.")
```

Comment:
**nit:** `403 Forbidden` is usually when a user is authenticated (i.e. the server knows who the user is), but the user does not have access to a specific resource (e.g. another user's docs or private data).

I think this would just be a general `400 Bad Request` error.